sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -78,7 +78,6 @@ abstract class Optimizer(catalogManager: CatalogManager)
     val operatorOptimizationRuleSet =
       Seq(
         // Operator push down
-        PushProjectionThroughUnion,
```

Comment:
This looks like a weird fix to me, by moving an optimizer rule from the main batch to a new batch before `Batch("Union", ...)`.

Can we fix `CombineUnions` instead? To support group-only `Aggregate`.

Reply:
+1